pkg/scheduler/flavorassigner/flavorassigner.go

```diff
@@ -437,14 +445,17 @@ func (a *FlavorAssigner) assignFlavors(log logr.Logger, counts []int32) Assignme
 		},
 	}
 
+	groupedRequests := make(map[string][]IndexedPodSet)
+	var groupsOrder []string
```

Comment:
What about creating a new data structure (may just use these two under the hood). So that the caller doesn't have to update both.

Reply:
I like this idea, mainly because it allows to offload the logic here with technical details.

We could approach it:
1. generically, like `InsertionOrderMultimap` 
2. specific, just a new file in the package, and maybe name `podSetGroups`

I'm leaning (only slightly) for (2.) as I don't know any other places where we could reuse it for now.

I'm ok in this PR, or to open an issue for follow up.

Reply:
I followed the (2.) and introduced a new file `podset_groups.go` with struct that hides the orderd map complexity.